llvm/lib/CodeGen/StackProtector.cpp

```diff
@@ -541,8 +540,7 @@ static bool CreatePrologue(Function *F, Module *M, Instruction *CheckLoc,
   AI = B.CreateAlloca(PtrTy, nullptr, "StackGuardSlot");
 
   Value *GuardSlot = getStackGuard(TLI, M, B, &SupportsSelectionDAGSP);
-  B.CreateCall(Intrinsic::getOrInsertDeclaration(M, Intrinsic::stackprotector),
-               {GuardSlot, AI});
+  B.CreateIntrinsic(Intrinsic::stackprotector, {}, {GuardSlot, AI});
```

Comment:
Looks like we could use an overload where we do not need to specify Types or RetTy (for non-overloaded intrinsics).

Reply:
Yeah, there could certainly be more useful overloads, and maybe more default arguments. But I didn't really want to do that in this patch.